.github/workflows/test.yml

```diff
@@ -26,6 +26,8 @@ jobs:
       run: |
         pip install -U pip setuptools wheel
         pip install -e .
+        # Workaround until Mypy regression is fixed.
+        pip install mypy==1.5.1
```

Comment:
This is unrelated to the other changes I made.
Current `master` branch is failing with latest version of Mypy for a reason I'm not sure to understand.

Reply:
Can you please provide a traceback?

Reply:
@sobolevn Sure:

```bash
$ pytest pytest_mypy_plugins/tests/test-mypy-config.yml -vvv
============================================================================== test session starts ==============================================================================
platform linux -- Python 3.11.5, pytest-7.4.3, pluggy-1.3.0 -- /home/delgan/programming/pytest-mypy-plugins/env/bin/python
cachedir: .pytest_cache
rootdir: /home/delgan/programming/pytest-mypy-plugins
configfile: pyproject.toml
plugins: mypy-plugins-3.0.0
collected 1 item                                                                                                                                                                

pytest_mypy_plugins/tests/test-mypy-config.yml::custom_mypy_config_strict_optional_true_set FAILED

=================================================================================== FAILURES ====================================================================================
__________________________________________________________________ custom_mypy_config_strict_optional_true_set __________________________________________________________________
/home/delgan/programming/pytest-mypy-plugins/pytest_mypy_plugins/tests/test-mypy-config.yml:4: 
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Output is not expected: 
E   Actual:
E     ../../home/delgan/programming/pytest-mypy-plugins/env/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi:154: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc] (diff)
E   Expected:
E     (empty)
============================================================================ short test summary info ============================================================================
FAILED pytest_mypy_plugins/tests/test-mypy-config.yml::custom_mypy_config_strict_optional_true_set - 
=============================================================================== 1 failed in 3.14s ===============================================================================
```

This can be reproduced by executing `mypy --no-silence-site-packages --no-strict-optional` on the following file:

```python
from typing import Optional
a: Optional[int] = None
a + 1  # should not raise an error
```

Mypy v1.6.0+ flags this snippet as invalid and exits with non-zero:

```
env/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi:154: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
Found 1 error in 1 file (checked 1 source file)
```

This doesn't happen when ` --no-silence-site-packages` is omitted.

Reply:
`--no-strict-optional` is not tested much in mypy (and for a good reason - it should not be used).

How does it affect us here?

Reply:
It is used to verify that inline `mypy_config` works as expected: https://github.com/typeddjango/pytest-mypy-plugins/blob/master/pytest_mypy_plugins/tests/test-mypy-config.yml

We can probably test it with another option, though.

Reply:
I stumbled upon this issue because tests failed with mypy-1.8.0 (https://bugs.gentoo.org/921901). Has the regression been reported upstream?